components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLArtifactResolveServlet.java

```
@@ -0,0 +1,290 @@
+            try {
+                MessageFactory messageFactory = MessageFactory.newInstance();
+                InputStream inStream = req.getInputStream();
+                SOAPMessage soapMessage = messageFactory.createMessage(new MimeHeaders(), inStream);
+                SOAPBody soapBody = soapMessage.getSOAPBody();
+                Iterator iterator = soapBody.getChildElements();
+
+                while (iterator.hasNext()) {
+                    SOAPBodyElement artifactResolveElement = (SOAPBodyElement) iterator.next();
+
+                    if (StringUtils.equals(SAMLConstants.SAML20P_NS, artifactResolveElement.getNamespaceURI()) &&
+                            StringUtils.equals(ArtifactResolve.DEFAULT_ELEMENT_LOCAL_NAME,
+                                    artifactResolveElement.getLocalName())) {
+
+                        id = URLDecoder.decode(artifactResolveElement.getAttribute("ID"),
+                                StandardCharsets.UTF_8.name());
+                        issueInstant = URLDecoder.decode(artifactResolveElement.getAttribute("IssueInstant"),
+                                StandardCharsets.UTF_8.name());
+
+                        SOAPBodyElement issuerElement = (SOAPBodyElement) artifactResolveElement.getFirstChild();
+                        SOAPBodyElement artifactElement = (SOAPBodyElement) issuerElement.getNextSibling();
+                        samlArtifact = URLDecoder.decode(artifactElement.getFirstChild().getNodeValue(),
+                                StandardCharsets.UTF_8.name());
+                    }
+                }
+            } catch (SOAPException e) {
+                throw new ServletException("Invalid SAML Artifact Resolve request received.", e);
+            }
+
+            if (samlArtifact != null) {
+                handleArtifact(req, resp, samlArtifact, id, issueInstant);
+            } else {
+                log.error("Invalid SAML Artifact Resolve request received.");
```

This will fail silently. Is that the expected behavior here.
Shouldn't the erroneous case be indicated back to the requestor